lib/modules/team_stats.rb

```diff
@@ -53,4 +53,48 @@ def favorite_opponent(team_id)
     end
     team.team_name
   end
-end 
+
+  def rival(team_id)
+  #Name of the opponent that has the highest win percentage against the given team.
+    relevant_games = @games.find_all do |game|
+      game.home_team_id == team_id || game.away_team_id == team_id
+    end
+
+    games_played = Hash.new(0)
+    #{opponent_id => times opponent played target_team}
+    relevant_games.map do |game|
+      if team_id == game.home_team_id
+        opponent = game.away_team_id
+        games_played[opponent] += 1
+      else
+        opponent = game.home_team_id
+        games_played[opponent] += 1
+      end
+    end
+
+    wins = Hash.new(0)
+    #{oppenent_id => times that target_team has won against opponent}
+    relevant_games.map do |game|
+      if team_id == game.home_team_id && game.home_goals > game.away_goals
+        #wins[team_id] += 1
+        #commented out because adding up all of team_id's wins moves it to the top of the list. Not what we want. 
```

Comment:
remove commented out old code and remove the conditional statements that are not being used